### PR TITLE
WIP 3.1.0 release (parallelization)

### LINKED ISSUE #781: 3.1.0 Release: Document how to run tests in parallel
- Add parallelization as a feature to any of the 'marketing' type docs
- Update CLI commands to include flags: https://cacoo.com/diagrams/1XyZbeou1rRyrEph#11C4D
- Update Dashboard to doc Parallel Calculator

### PR DESCRIPTION
**Pull request by jennifer-shehane**
- closes #781

### CODE REVIEW

File: source/guides/guides/continuous-integration.md [hunk 1]
```diff
@@ -44,13 +52,13 @@ Depending on which CI provider you use, you may need a config file. You'll want
 
 ## Caching the Cypress Binary
 
-As of Cypress version 3.0, Cypress downloads its binary to the global system cache - on linux that's `~/.cache/Cypress`. Ensuring this cache persists across builds, you can shave minutes off install time by preventing a large binary download.
+As of {% url "Cypress version 3.0" changelog#3-0-0 %}, Cypress downloads its binary to the global system cache - on linux that is `~/.cache/Cypress`. Ensuring this cache persists across builds, you can shave minutes off install time by preventing a large binary download.
```
Comment (richdouglasevans) [inline source/guides/guides/continuous-integration.md]: `Minor`

The second sentence reads a wee bit "off"... possibly, might just be me 🤔

Maybe...

```
By ensuring this cache persists across builds you can shave minutes off install time by preventing a large binary download.
```

File: source/guides/guides/command-line.md [hunk 2]
```diff
@@ -287,10 +319,10 @@ cypress cache clear
 You can use also require Cypress as a Node module.
 
 {% note warning %}
-The Cypress module is brand new and we are still adding more functionality to it. Its API may change in the future.
+The Cypress module is still in beta and we are still adding more functionality to it. Its API may change in the future.
 {% endnote %}
 
-Here's an example of programmatically running a spec file:
+Here is an example of programmatically running a spec file:
```
Comment (richdouglasevans) [inline source/guides/guides/command-line.md]: Well darn. I just added a contraction in another PR 😢 I'll respect this house style in the event of future PRs, sorry.

File: source/guides/guides/parallelization.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+{% note info %}
+# {% fa fa-graduation-cap %} What You'll Learn
+
+- 
+{% endnote %}
+
+# Overview
+
+If your project has a large number of tests, it can take a large amount of time for tests to complete running when run on one machine serially. Spreading your tests across many virtual machines and running tests in parallel can save your team time and money when running tests in Continuos Integration.
```
Comment (richdouglasevans) [inline source/guides/guides/parallelization.md]: `Typo`

Second to last word: `Continuos` ➡️ `Continuous`

File: source/guides/guides/parallelization.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+Running tests in parallel requires the `--record` flag be passed. This ensures Cypress can properly collect the data needed to parallelize the run. This also gives you the full benefit of seeing the results of the parallelization in our Dashboard Service. If you have not set up your project to record, check out our Set up guide.
+
+If you've passed the `--parallel` flag, Cypress is now responsible for a few things:
+
+- We automatically look up the number of available machines.
+- We split up all spec files within your project across any available machines.
+
+# Balance strategy
+
+Cypress will automatically balance your spec files across the available machines in your CI provider. Cypress calculates which spec file to run based on from data collected from previous runs. This ensures that your spec files run as fast as possible, with no need for manual configuration or tweaking.
```
Comment (richdouglasevans) [inline source/guides/guides/parallelization.md]: `Typo`

`based on from data` ➡️ `from data`

File: source/guides/guides/parallelization.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+You'll want to refer to your CI provider's documentation on how to set up multiple machines to run in your CI environment first.
+
+Once multiple machines are available within your CI environment, you can pass the {% url "`--parallel`" command-line#cypress-run-parallel %} key to {% url "`cypress run`" command-line#cypress-run %} to have your recorded tests parallelized.
+
+```shell
+cypress run --record --parallel
+```
+
+Running tests in parallel requires the `--record` flag be passed. This ensures Cypress can properly collect the data needed to parallelize the run. This also gives you the full benefit of seeing the results of the parallelization in our Dashboard Service. If you have not set up your project to record, check out our Set up guide.
```
Comment (richdouglasevans) [inline source/guides/guides/parallelization.md]: Consider hyperlinking to the set up guide to make navigating to the next thing easier.

Comment (richdouglasevans) [reply source/guides/guides/parallelization.md]: I suspect a PR for the documentation ain't the right place for this comment... but might Cypress default to turning reporting on when the parallel flag is set? Rationale: being explicit is often good but sensible defaulting is too.

Dunno: I'll watch this space, no biggie.

File: source/guides/references/changelog.md [hunk 6]
```diff
@@ -2,6 +2,29 @@
...
+**Summary:**
+
+- Cypress can now run recorded tests in parallel across multiple virtual machines. Fixes {% issue 1690 %} and {% issue 64 %}.
+
+**Features:**
+
+- `cypress run` accept a new `--parallel` flag to run specs in parallel.
+- `cypress run` accepts a new `--group` flag to group multiple spec files within a single run. Fixes {% issue 2153 %}
+- `cypress run` accepts a new `--ci-build-id` flag.
+- The Dashboard Service has new "Insights" available within each run. You can now see how the run was parallelized and calculate how adding or removing virtual machines would change future runtime.
```
Comment (richdouglasevans) [inline source/guides/references/changelog.md]: This all sounds most excellent! Looking forward to it: good work Team Cypress! 👍

File: source/guides/guides/parallelization.md (hunk 4 above)
Comment (richdouglasevans) [reply source/guides/guides/parallelization.md]: The parallelization seems simple and powerful: good docs here Jennifer 👍 

> If these "drive by" reviews aren't cool -- I know this is a WIP -- then do say and I'll stop. We're on a Cypress roll at the moment at work so I'm all over all things Cypress for a wee while.

File: source/guides/guides/parallelization.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+- 
+{% endnote %}
+
+# Overview
+
+If your project has a large number of tests, it can take a large amount of time for tests to complete running when run on one machine serially. Spreading your tests across many virtual machines and running tests in parallel can save your team time and money when running tests in Continuos Integration.
+
+Cypress can run recorded tests in parallel across multiple virtual machines.
+
+This guide assumes you already have your project running and recording within Continuous Integration. If you haven't set up your project yet, check out our Continuous Integration guide.
```
Comment (richdouglasevans) [inline source/guides/guides/parallelization.md]: Consider hyperlinking to the CI guide: makes navigating these excellent docs even more of a pleasure.

```
check out our {% url 'Continuous Integration' continuous-integration %} guide.
```

File: source/guides/guides/parallelization.md (hunk 5 above)
Comment (jennifer-shehane) [reply source/guides/guides/parallelization.md]: We definitely do not want to record by default when the parallel flag is passed. 'recording' means that a select amount of your data is sent to our servers to deliver value in the Dashboard - and we'd hate for someone to think they are not sending us data when indeed they are.

**Pull request merged**